### PR TITLE
Add ActionCable support

### PR DESCRIPTION
**Pull request by tombruijn**
Support ActionCable by overriding the `#perform_action` method and
wrapping it in an AppSignal transaction. The `#perform_action` is called
by ActionCable once a message is send to the channel.

We also track subscribe and unsubscribe events as they can also contain
complex logic added by the user, which is something we want to track. To
track this two "around" callbacks are added to ActionCable on load.

I've added an example application here for you to test it against: https://github.com/appsignal/appsignal-examples/tree/rails-5%2Baction-cable

Transactions are tested with the data from the extension as per issue #252 

## Support

https://app.intercom.io/a/apps/yzor8gyw/respond/inbox/conversation/9011790920
https://app.intercom.io/a/apps/yzor8gyw/respond/inbox/conversation/9039706354

---

**Note:** Rebased continuation of #275, which I'll close after creating this. Rebased the branch and created a new PR, because I had to rebase it to clean up some really old work that was extracted to separate PRs and things that really weren't necessary anymore. I also didn't want to rebase the original PR and force push it, in case someone is relying on it, breaking their build.

### CODE REVIEW

**Comment by tombruijn** [comment]
Small update: Added instrumentation around callbacks so that they always have one action cable event.

File: spec/lib/appsignal/hooks/action_cable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+          expect(Appsignal.active?).to be_truthy
+          transaction
+
+          expect(Appsignal::Transaction).to receive(:create)
+            .with(request_id, Appsignal::Transaction::ACTION_CABLE, kind_of(ActionDispatch::Request))
+            .and_return(transaction)
+          allow(Appsignal::Transaction).to receive(:current).and_return(transaction)
+          expect(transaction.ext).to receive(:complete) # and do nothing
+
+          # TODO: Nicer way to stub this without a websocket?
```
Comment (jeffkreeftmeijer) [inline spec/lib/appsignal/hooks/action_cable_spec.rb]: Could you remove this TODO and create an issue for it, so it doesn't get lost?

Comment (tombruijn) [inline spec/lib/appsignal/hooks/action_cable_spec.rb]: Ah yes, that TODO. I wanted to fix that before submitting it, but looks like it slipped through. I'll have a looksy

Comment (tombruijn) [inline spec/lib/appsignal/hooks/action_cable_spec.rb]: Explained the comment a bit better, but I'll leave the stub like this for now.

**Comment by tombruijn** [comment]
Looks like this doesn't support standalone mode at the moment: http://edgeguides.rubyonrails.org/action_cable_overview.html#standalone

**Comment by rmachielse** [comment]
👌  @tombruijn thanks a lot for this

**Comment by p0wl** [comment]
very nice 🎉

**Pull request merged**